### ISSUE TITLE
mod_proxy_http2 authority / host header mismatch

### ISSUE THREAD

**Issue by szbalint** [icing/mod_h2#124]
When trying to use Apache with mod_http2 and reverse proxying with mod_proxy_http2 into a backend server that can also talk HTTP/2, I seem to be running into an issue with mismatched :authority and host headers.

When I initiate a client connection with nghttp2, these are the clientside headers that are sent:

```
[  0.009] send HEADERS frame <length=66, flags=0x25, stream_id=13>
          ; END_STREAM | END_HEADERS | PRIORITY
          (padlen=0, dep_stream_id=11, weight=16, exclusive=0)
          ; Open new stream
          :method: GET
          :path: /foo/bar/
          :scheme: https
          :authority: testing.example.com
          accept: */*
          accept-encoding: gzip, deflate
          user-agent: nghttp2/1.12.0-DEV
```
but it seems to be that after mod_proxy_http2 tries to proxy this request back to a backend server (in this case nginx), the headers end up being:

```
*4 http2 add header to hpack table: ":authority: 127.0.0.1:82"
*4 http uri: "/foo/bar/"
(...)
*4 http2 add header to hpack table: "host: testing.example.com"
```

I could be getting this wrong, but as far as I know `.authority`  and `host` should be set to the same value.

A secondary/wishlist item might be that `ProxyPreserveHost` does not seem implemented for mod_proxy_http2 (or at least doesn't have an effect on either headers). Using `mod_proxy_http` to communicate via the backend server over HTTP/1.x is a viable workaround for me though.

Would it seem reasonable to get closer to the `mod_proxy_http` behaviour, where the host header is preserved/rewritten based on the `ProxyPreserveHost` configuration option? Plus the `:authority` header matches `host` in all cases?

Please don't hesitate to correct me if I'm wildly off track here and thank you in advance!

I'm using `v1.8.3-git, nghttp2 1.17.0`.

**Comment by icing** [comment]
@szbalint please try the just released v1.9.1 where ```ProxyPreserveHost``` is supported now.

**Comment by icing** [comment]
Fixed.
